pkg/document/remoteresolver/remoteresolver.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package remoteresolver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/trustbloc/edge-core/pkg/log"
+	"github.com/trustbloc/sidetree-core-go/pkg/document"
+
+	"github.com/trustbloc/orb/pkg/activitypub/client/transport"
+)
+
+var logger = log.New("remote-resolver")
+
+const (
+	didLDJson = "application/did+ld+json"
+)
+
+// Resolver resolves document from remote server.
+type Resolver struct {
+	httpClient httpClient
+}
+
+type httpClient interface {
+	Get(ctx context.Context, req *transport.Request) (*http.Response, error)
+}
+
+// New create new remote resolver.
+func New(httpClient httpClient) *Resolver {
+	return &Resolver{
+		httpClient: httpClient,
+	}
+}
+
+// ResolveDocumentFromResolutionEndpoints resolved document from resolution endpoints.
+func (rr *Resolver) ResolveDocumentFromResolutionEndpoints(id string, endpoints []string) (*document.ResolutionResult, error) { //nolint:lll
+	if len(endpoints) == 0 {
+		return nil, fmt.Errorf("must provide at least one remote resolver endpoint in order to retrieve data")
+	}
+
+	var errMsgs []string
+
+	for _, endpoint := range endpoints {
+		reqURL := fmt.Sprintf("%s/%s", endpoint, id)
+
+		responseBytes, err := rr.send(reqURL)
+		if err != nil {
+			errMsg := fmt.Sprintf("endpoint[%s]: %s", reqURL, err.Error())
+
+			errMsgs = append(errMsgs, errMsg)
+
+			continue
+		}
+
+		var respObj document.ResolutionResult
+
+		err = json.Unmarshal(responseBytes, &respObj)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal resolution result[%s] for remote request[%s]",
```

Comment:
In this case maybe the error can be added to errMsgs and continue to the next URL.

Reply:
sure, I will change this in my next PR